st2api/st2api/controllers/v1/actions.py

```diff
@@ -235,9 +236,51 @@ def delete(self, ref_or_id, requester_user):
             ref_or_id,
             action_db,
         )
+        
+        BASE_PATH = cfg.CONF.system.base_path
+        pack_name = action_db["pack"]
+        entry_point = action_db["entry_point"]
+        metadate_file = action_db["metadata_file"]
+        action_entrypoint_file_path = os.path.join(
+            BASE_PATH, "packs", pack_name, "actions", entry_point
+        )
+        action_metadata_file_path = os.path.join(
+            BASE_PATH, "packs", pack_name, metadate_file
+        )
 
         try:
             Action.delete(action_db)
+
+            if os.path.exists(action_entrypoint_file_path):
+                try:
+                    os.remove(action_entrypoint_file_path)
+                except PermissionError:
+                    LOG.error(
+                        'No permission to delete the "%s" file',
+                        action_entrypoint_file_path,
+                    )
+                except Exception as e:
+                    LOG.error(
+                        'Unable to delete "%s" file. Exception was "%s"',
+                        action_entrypoint_file_path,
+                        e,
+                    )
+
+            if os.path.exists(action_metadata_file_path):
+                try:
+                    os.remove(action_metadata_file_path)
+                except PermissionError:
+                    LOG.error(
+                        'No permission to delete the "%s" file',
+                        action_metadata_file_path,
+                    )
+                except Exception as e:
+                    LOG.error(
+                        'Unable to delete "%s" file. Exception was "%s"',
+                        action_metadata_file_path,
+                        e,
+                    )
+
```

Comment:
Can you create a function at https://github.com/StackStorm/st2/blob/master/st2common/st2common/services/packs.py to delete the action entrypoint and action metadata files? Move the path calculation that you did above into this function as well.